src/core/crypto/crux.rs

```diff
@@ -125,12 +141,21 @@ impl DataDecrypter<DecryptedDataGroup> for EncryptedDataGroup {
         self,
         state: &AppState,
         identifier: &Identifier,
+        custodian: Custodian,
     ) -> errors::CustomResult<DecryptedDataGroup, errors::CryptoError> {
         let version = FxHashSet::from_iter(self.0.values().map(|d| d.version));
         let decrypted_keys = Key::get_multiple_keys(state, identifier, version)
             .await
             .switch()?;
 
+        let mut stored_tokens = decrypted_keys.values().map(|k| &k.token);
```

Comment:
```suggestion
        let stored_tokens = decrypted_keys.values().map(|k| &k.token);
```

Reply:
It's required, as it's a opaque type that will be acted upon: when we call `.all` on it